src/functionalTest/java/uk/gov/hmcts/reform/pip/account/management/controllers/InactiveAccountsManagementTest.java

```diff
@@ -0,0 +1,277 @@
+
+        assertThat(response.getStatusCode()).isEqualTo(OK.value());
+
+        return response.getBody().as(MediaApplication.class);
+    }
+
+    private AzureAccount createAzureAccount(String email, Roles role) {
+        String requestBody = """
+            [
+                {
+                    "email": "%s",
+                    "firstName": "%s",
+                    "surname": "%s",
+                    "role": "%s",
+                    "displayName": "%s"
+                }
+            ]
+            """.formatted(email, FIRST_NAME, SURNAME, role, TEST_NAME);
+
+
+        Response response = doPostRequest(ADD_USER_B2C_URL, headers, issuerId, requestBody);
+        List<AzureAccount> azureAccountList = OBJECT_MAPPER.convertValue(
+            response.jsonPath().getJsonObject("CREATED_ACCOUNTS"),
+            new TypeReference<>() {
+            }
+        );
+        assertThat(azureAccountList.size()).isEqualTo(1);
+
+        return azureAccountList.get(0);
+    }
+
+    private PiUser createSystemAdminAccount(String email) {
+        String requestBody = """
+            {
+                "email": "%s",
+                "firstName": "%s",
+                "surname": "%s"
+            }
+            """.formatted(email, FIRST_NAME, SURNAME);
+
+
+        Response response = doPostRequest(ADD_SYSTEM_ADMIN_B2C_URL, headers, issuerId, requestBody);
+        PiUser piUser = response.getBody().as(PiUser.class);
+        assertThat(response.getStatusCode()).isEqualTo(OK.value());
+
+        return piUser;
+    }
+
+    private String createUser(String email,
+                              String provenancesId, Roles role, UserProvenances provenances) {
+        String requestBody = """
+            [
+                {
+                    "email": "%s",
+                    "provenanceUserId": "%s",
+                    "roles": "%s",
+                    "userProvenance": "%s"
+                }
+            ]
+            """.formatted(email, provenancesId, role, provenances);
+
+        Response postResponse = doPostRequest(ADD_PI_USER_URL, headers, issuerId, requestBody);
+        List<UUID> piUsersList = OBJECT_MAPPER.convertValue(
+            postResponse.jsonPath().getJsonObject("CREATED_ACCOUNTS"),
+            new TypeReference<>() {
+            }
+        );
+        assertThat(piUsersList.size()).isEqualTo(1);
+        return piUsersList.get(0).toString();
+    }
+
+    void updateUserAccountLastVerifiedDate(String userProvenancesId, UserProvenances userProvenances,
+                                           Map<String, String> fieldsToUpdate) {
+        JSONObject jsonFieldsToUpdate = new JSONObject(fieldsToUpdate);
+
+        Response response = doPutRequestWithJsonBody(UPDATE_USER_INFO + "/" + userProvenances + "/" + userProvenancesId,
+                                              headers, jsonFieldsToUpdate.toString());
+
+        assertThat(response.getStatusCode()).isEqualTo(OK.value());
+    }
+
```

Comment:
For the Delete tests, I think we need to do a GET request to check the user no longer exists in the DB

Reply:
added.